File: pop-booking-api/src/dk/knet/pop/booking/models/BookableItem.java
package dk.knet.pop.booking.models;

import javax.persistence.*;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.ColumnDefault;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Entity
@Builder
@Table(name = "bookable_item")
public class BookableItem {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	private String name;
	private BookingType bookingType; // this is currently not used
	private String color;

	@Column(nullable = false)
	private Integer maxBookableHours;

	@Column(nullable = false) @ColumnDefault(value = "0")
	private Status status = Status.ACTIVE;
	private String statusMessage;

	public enum Status {
		OUT_OF_ORDER, ACTIVE
	}

}

File: pop-booking-web/src/models/BookableItem.js
import {computed, decorate, observable} from "mobx";
import {D} from "../D";

export default class BookableItem {


    id;

    color;
    bookingType;
    maxBookableHours;
    status;
    statusMessage;
    store;

    _name;

    constructor(json) {
        this.id = json.id;
        this._name = json.name;
        this.bookingType = json.bookingType || null;
        this.color = json.color;
        this.maxBookableHours = json.maxBookableHours;
        this.status = json.status;
        this.statusMessage = json.statusMessage;
    }

    get name() {
        return D(this._name);
    }

    asJson = () => {
        return {
            id: this.id,
            color: this.color,
            bookingType: this.bookingType,
            maxBookableHours: this.maxBookableHours,
            name: this._name,
            status: this.status,
            statusMessage: this.statusMessage
        }
    }
}

decorate(BookableItem, {
    name: computed,
    color: observable,
    status: observable,
    statusMessage: observable,
    maxBookableHours: observable
})